astro/src/content/blog/building-self-hostable-application.mdx

```diff
@@ -0,0 +1,366 @@
+
+This includes detailed documentation around:
+
+* installation
+* scaling
+* performance tuning
+* monitoring
+* troubleshooting
+* system requirements
+* infrastructure integrations
+* tooling integrations
+* upgrades
+
+You'll need to make this as thorough, accurate and complete as you can. You'll also need to clearly tag them with versions of your software, because capabilities and requirements change as your software evolves.
+
+The number of possible combinations of how your software can be installed or operated is nearly infinite. Try to limit the effort by documenting common integrations and complementary software components.
+
+Clearly document your changes too, as part of your release notes. You can do this in lo-fi ways such as semver or high fidelity ways like old-style release notes. 
+
+### Configuration Management
+
+The type of organization that can confidently self-host critical infrastructure also leverage configuration management. They want to configure your application via an API or tooling like Terraform or Pulumi. 
+
+Offering this functionality allows your customers to manage changes to your product in the same way they manage other parts of their application infrastructure.
+
+### Professional Services
+
+Ah, the bane of every software company's margin. Professional services can help your customer implement your product quickly and effectively.
+
+If you don't want to build out your own professional services organization, build relationships with consulting companies. Be aware that both parties want the other one to bring them customers and make them money, so these relationships can be difficult to kickstart.
+
+## What Kind Of Software Makes Sense
+
+Not every type of developer tool makes sense to build as a self-hostable solution. But if your product:
+
+* has an offline use case
+* appeals to customers familiar with operating software
+* solves a critical problem
+* integrates with existing well known components
+* offers functionality used by a large number of applications
+
+Then you should consider creating a self-hostable solution.
+
+## Monetization 
+
+If you want to sell your self-hostable product, you have a few options.
+
+* support
+* advanced features
+* hosting
+
+Each of these have strengths and weaknesses.
+
+Support is easy to offer; allow some users to pay for access to a slack channel, ticketing system or email address. However, support requires an investment in employees and doesn't scale linearly. For a self-hostable product, support can be more complex to offer, as discussed above.
+
+Advanced features can scale revenue without increasing costs, but require significant upfront investment to build out. They also require a licensing system. This typically requires network access, which removes the offline benefits.
```

Comment:
You mention licensing here, so I think under the Monetization section you should say something like: "in addition to licensing..." or add it as a bullet point.